test/view/linear_distribute.cpp

```diff
@@ -0,0 +1,60 @@
+// Range v3 library
+//
+//  Copyright Eric Niebler 2014
+//  Gonzalo Brito Gadeschi 2017
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+#include <iostream>
+#include <cmath>
+#include <range/v3/algorithm/equal.hpp>
+#include <range/v3/view/linear_distribute.hpp>
+#include <range/v3/view/transform.hpp>
+#include "../simple_test.hpp"
+#include "../test_utils.hpp"
+
+int main() {
+    using ranges::view::linear_distribute;
+
+    auto float_eq = [](double a, double b) {
+        return (int)std::floor(std::abs(a - b)) == 0;
+    };
+
+    {
+        auto irng = linear_distribute(0, 1, 2);
+        static_assert(ranges::RandomAccessRange<decltype(irng)>(), "");
+        static_assert(ranges::SizedRange<decltype(irng)>(), "");
+        check_equal(irng, {0, 1});
+    }
+    {
+        auto irng = linear_distribute(1, 3, 3);
+        check_equal(irng, {1, 2, 3});
+    
+    }
+    {
+        auto irng = linear_distribute(0, 21, 22);
+        auto frng = linear_distribute(0., 21., 22);
+        check_equal(irng, frng | ranges::view::transform([](double v) { return int(v); }));
+    }
+    {
+        auto frng = linear_distribute(0.0, 1.0, 11);
+        static_assert(ranges::InputRange<decltype(frng)>(), "");
+        static_assert(ranges::SizedRange<decltype(frng)>(), "");
+        CHECK(ranges::equal(frng, {0., 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0},
+                            float_eq));
+    }
+    {
+        auto frng = linear_distribute(1.0, 3.0, 21);
+        CHECK(ranges::equal(frng,
+                            {1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9,
+                             2.0, 2.1, 2.2, 2.3, 2.4, 2.5, 2.6, 2.7, 2.8, 2.9,
+                             3.0},
+                            float_eq));
+    }
```

Comment:
You need to test the `to==from && n==1` case, for both integral and floating point types. And I'm curious what happens in the `to==from && n>1` case.